source/io/module_javaScriptInvoke.js

```diff
@@ -349,65 +348,26 @@
             REJECTED: 'REJECTED'
         };
 
-        var updateReturnValue = function (
-            functionName,
-            returnTypeName,
-            returnValuePointer,
-            returnUserValue,
-            errorValueRef) {
-            if (returnTypeName === 'StaticTypeAndData') {
-                // User doesn't want return value. If we're passing '*' for the return in VIA code, we get StaticTypeAndData
+        var updateReturnValue = function (functionName, returnValueRef, returnUserValue, errorValueRef) {
+            if (returnValueRef.dataRef === 0) {
+                // User doesn't want to return value. We get here, if we're passing '*' for return parameter in VIA code
                 return;
             }
-
-            var typeConfig = typeFunctions[returnTypeName];
-            var newError = {
-                status: true
+            var valueAndError = {
+                userValue: returnUserValue,
+                errorValueRef: errorValueRef,
+                functionName: functionName
             };
-
-            if (typeConfig === undefined) {
```

Comment:
What is the jsInvokePeeker behavior when dealing for a type if doesn't know? This is the behavior this validation is doing. Can we incorporate it in the jsInvokePeeker?
Note: This is mostly for handwritten VIA. DFIR doesn't generate types that we don't know.

Reply:
The reflect API will throw exception which will turn into an unsupported error.